Classes/Plugin/PluginBase.php

```diff
@@ -238,12 +246,12 @@ public function pi_loadLL()
             $basePath = 'EXT:' . $this->extKey . '/Resources/Private/Language/Plugin' . $languageFileName . '.xml';
 
             // Read the strings in the required charset (since TYPO3 4.2)
-            $this->LOCAL_LANG = GeneralUtility::readLLfile($basePath,
-                $this->LLkey, $GLOBALS['TSFE']->renderCharset);
+            $this->LOCAL_LANG = $this->languageFactory->getParsedData($basePath,
+                $this->LLkey, $GLOBALS['TSFE']->renderCharset, 3);
             $alternativeLanguageKeys = GeneralUtility::trimExplode(',',
                 $this->altLLkey, true);
             foreach ($alternativeLanguageKeys as $languageKey) {
-                $tempLL = GeneralUtility::readLLfile($basePath, $languageKey);
+                $tempLL = $this->languageFactory->getParsedData($basePath, $languageKey, $GLOBALS['TSFE']->renderCharset, 3);
```

Comment:
I would keep it, because in the core it is used like this, even when everybody should allready use UTF-8.